From 233e1a436022ba83fe43e487da6fb5992d8ee08c Mon Sep 17 00:00:00 2001
From: Anthony Roberts <anthony.roberts@linaro.org>
Date: Mon, 25 Jul 2022 10:40:25 +0100
Subject: [PATCH 5/5] Add check for clang on Windows ARM64 platforms

This commit adds a warning/error for ARM64 platforms in the event that
clang is chosen as the compiler.
---
 cmake/os/Windows.cmake | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/cmake/os/Windows.cmake b/cmake/os/Windows.cmake
index 78a363871f0..a2a4c666501 100644
--- a/cmake/os/Windows.cmake
+++ b/cmake/os/Windows.cmake
@@ -32,17 +32,25 @@ INCLUDE (CheckCSourceRuns)
 INCLUDE (CheckSymbolExists)
 INCLUDE (CheckTypeSize)
 
+IF(CMAKE_GENERATOR_PLATFORM STREQUAL "ARM64")
+  SET(WIN_ARM64 1)
+ENDIF()
+
 IF(MY_COMPILER_IS_CLANG)
+  IF(WIN_ARM64)
+    IF(FORCE_UNSUPPORTED_COMPILER)
+      MESSAGE(WARNING "Using clang for ARM64 platforms is untested!")
+    ELSE()
+      MESSAGE(WARNING "Clang on windows ARM64 platforms is currently unsupported")
+    ENDIF()
+  ENDIF()
+
   SET(WIN32_CLANG 1)
   SET(CMAKE_INCLUDE_SYSTEM_FLAG_C "/imsvc ")
   SET(CMAKE_INCLUDE_SYSTEM_FLAG_CXX "/imsvc ")
   ADD_DEFINITIONS(-DWIN32_CLANG)
 ENDIF()
 
-IF(CMAKE_GENERATOR_PLATFORM STREQUAL "ARM64")
-  SET(WIN_ARM64 1)
-ENDIF()
-
 # avoid running system checks by using pre-cached check results
 # system checks are expensive on VS since every tiny program is to be compiled
 # in a VC solution.
-- 
2.35.1.windows.2

